coding/python_style_guide.rst

```diff
@@ -12,648 +13,671 @@ To request changes to these standards, please file an :ref:`RFC <decision-making
 
-User-defined names with double leading and trailing underscores MUST NOT be used
---------------------------------------------------------------------------------
+Code MAY be validated with flake8
+---------------------------------
 
-Names with double leading and trailing underscores are reserved by Python (e.g. ``__init__``, ``__name__``, ``__str__``).
+The flake8_ tool may be used to validate Python source code against the portion of PEP 8 adopted by Data Management.
+In addition, flake8_ statically checks Python for code errors.
+The separate `pep8-naming`_ plugin validates names according to the DM Python coding style.
 
-.. _style-guide-py-2-2:
+.. note::
+   
+   Flake8 only validates code against PEP 8 specifications, but does not check the full coding standard listed here.
 
-User defined names SHOULD NOT shadow python built-in functions
---------------------------------------------------------------
+.. _flake8: https://flake8.readthedocs.io
+.. _pep8-naming: http://pypi.python.org/pypi/pep8-naming
 
-Names which shadow a python built-in function may cause confusion for readers of the code.
-Creating a more specific identifier is suggested to avoid collisions.
-In the case of *filter*, ``filterName`` may be appropriate; for *filter objects*, something like ``filterObj`` might be appropriate.
+Flake8 installation
+^^^^^^^^^^^^^^^^^^^
 
-.. _style-guide-py-2-3:
+Linters are installable with :command:`pip`:
 
-Class names SHOULD use CamelCase with leading uppercase
--------------------------------------------------------
+.. code-block:: bash
 
-Python class names should follow the same conventions as C++ class names---they should be ``CamelCase`` with leading uppercase.
-Note that exceptions are classes and thus follow the same convention.
+   pip install flake8
+   pip install pep8-naming
 
-.. _style-guide-py-2-4:
+Flake8 command line invocation
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-Exception names relating to errors SHOULD include the suffix Error
-------------------------------------------------------------------
+.. code-block:: bash
 
-An error exception name SHOULD include the suffix ``Error``.
+   flake8 --ignore=E133,E226,E228,N802,N803 --max-line-length=110 {{python_dir}}
 
-.. _style-guide-py-2-5:
+where ``{{python_dir}}`` is a directory with Python source files.
 
-Class Attribute Names SHOULD be camelCase with leading lowercase
-----------------------------------------------------------------
+Flake8 configuration files
+^^^^^^^^^^^^^^^^^^^^^^^^^^
 
----
+LSST DM Packages may also include a :file:`setup.cfg` file with `PEP 8`_ exceptions:
 
-.. _style-guide-py-2-6:
+.. code-block:: ini
 
-Module methods (free functions) SHOULD be camelCase with leading lowercase
---------------------------------------------------------------------------
+	[flake8]
+	max-line-length = 110
+	ignore = E133, E226, E228, E251, N802, N803
 
----
+:command:`flake8` can be invoked without arguments when this configuration is present.
```

Comment:
The trick is the inconsistent behavior that this can lead to if people forget which packages have them and which don't.